wicket-core/src/main/java/org/apache/wicket/markup/head/JavascriptReferenceType.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.wicket.markup.head;
+
+public enum JavascriptReferenceType {
+
+	TEXT_JAVASCRIPT("text/javascript"), MODULE("module");
```

Comment:
What about other types like `importmap` ?
Using an enum makes it hard to use custom values when/if needed.

Reply:
Hi Martin, 

thank you for your remark.
I just started with the approach using a simple String instead of an Enum.
But then I realized, that there are only two "real" types for Javascript-References: Javascript-Mimetype ( "text/javascript", "application/javascript", "application/ecmascript") and "module".
All other types imply that the src-Attribute is ignored and the "real content" is placed inside the body of the <script>-Tag. But this is not the intention of the AbstractJavaScriptReferenceHeaderItem.
The documentation I used: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type

Kind regards
Hans